dolphin/drivers/manager.py

```diff
@@ -35,9 +42,45 @@ def get_storage_registry():
         """Show register parameters which the driver needs."""
         pass
 
+    @coordination.synchronized('driver-{register_info[vendor]}-'
+                               '{register_info[model]}')
     def register_storage(self, context, register_info):
-        """Discovery a storage system with register parameters."""
-        pass
+        """Discovery a storage system with access information."""
+        # Check same access info from DB
+        access_info = copy.deepcopy(register_info)
+        vendor, model = access_info.pop('vendor'), access_info.pop('model')
+        db_access_info = db.access_info_get_all(context, sort_keys=['host'],
+                                                filters=access_info)
+        if db_access_info:
+            msg = _("Storage device has been registered.")
+            raise exception.Conflict(msg)
+
+        # Load and initialize a driver
+        # todo: add exception handler
+        driver = stevedore.driver.DriverManager(
+            namespace=self.NAMESPACE,
+            name='%s %s' % (vendor, model),
+            invoke_on_load=True
+        ).driver
+
+        storage = driver.register_storage(context,
+                                          register_info)
+        if storage:
+            storage_id = six.text_type(uuidutils.generate_uuid())
+            access_info['storage_id'] = storage_id
+            # todo
+            # access_info['password'] = cryptor.encode(
+            #     access_info['password'])
+            db.access_info_create(context, access_info)
```

Comment:
any reason to messup the current architecture? I mean any specififc reason to add database creation, access logic to drivermanager?.  In this another thirdparty controller can not call our drivermanager.

Reply:
it's inevitable that the driver manager need to access the db, such as the driver or driver manager get the access info from db. we can not pass or save access info in memory because it contains the credential data.

Reply:
I try to move db access to api layer first.

Reply:
Seems no consistency mechanism here, need to consider the situation that access_info created in db successfully, but storage created failed.

Reply:
Thanks, I will concern about it.